server/util.c

```diff
@@ -2201,6 +2201,219 @@ AP_DECLARE(char *) ap_escape_html2(apr_pool_t *p, const char *s, int toasc)
     x[j] = '\0';
     return x;
 }
+
+/* TODO: not sure about copyright&license of below code */
+
```

Comment:
Digging further this appears to be MIT licensed.

Reply:
One possible thing to do is rework the escaping code from APR, which is Apache licensed and already belongs to the ASF:

https://github.com/apache/apr-util/blob/1.7.x/json/apr_json_encode.c#L63

Reply:
As explained in the added code for using apr_json, I'm not sure it's worth the added complexity here.